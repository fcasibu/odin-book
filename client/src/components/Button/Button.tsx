import * as React from "react";

const padding = {
    sm: "px-2 py-1",
    md: "px-4 py-2",
    lg: "px-6 py-4",
};

const colors = {
    success: "bg-green-500",
    default: "bg-sky-500",
};

type Size = "sm" | "md" | "lg";

type Variant = "success" | "default";

type Props = {
    children: React.ReactNode;
    size: Size;
    variant: Variant;
};

const Button = ({ children, size, variant }: Props) => {
    return (
        <button
            className={`${colors[variant]} text-white font-bold rounded-md ${padding[size]} hover:scale-95 active:scale-90 transition-all ease-linear`}
        >
            {" "}
            {children}
        </button>
    );
};

export default Button;
