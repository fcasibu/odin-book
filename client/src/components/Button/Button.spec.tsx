import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Button from "./Button";

describe("Button", () => {
    it("should render and match the snapshot", () => {
        const { container } = render(
            <Button variant="success" size="sm">
                Hello World!
            </Button>
        );

        expect(container).toMatchSnapshot();
    });

    it("should render the correct sizes and colors", () => {
        const { rerender } = render(
            <Button variant="default" size="sm">
                Hello World!
            </Button>
        );

        const button = screen.getByRole("button");
        expect(button).toHaveClass("bg-sky-500 px-2 py-1");

        rerender(
            <Button variant="success" size="md">
                Hello World!
            </Button>
        );

        expect(button).toHaveClass("bg-green-500 px-4 py-2");

        rerender(
            <Button variant="default" size="lg">
                Hello World!
            </Button>
        );

        expect(button).toHaveClass("bg-sky-500 px-6 py-4");
    });
});
