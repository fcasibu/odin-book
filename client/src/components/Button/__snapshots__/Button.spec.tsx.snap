// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render and match the snapshot 1`] = `
<div>
  <button
    class="bg-green-500 text-white font-bold rounded-md px-2 py-1 hover:scale-95 active:scale-90 transition-all ease-linear"
  >
     
    Hello World!
  </button>
</div>
`;
